gateway/reverse_proxy.go

```diff
@@ -1220,13 +1220,17 @@ func (p *ReverseProxy) WrappedServeHTTP(rw http.ResponseWriter, req *http.Reques
 	}
 
 	p.addAuthInfo(outreq, req)
+	err := p.addUpstreamOAuthInfo(outreq, req)
```

Comment:
addAuthInfo is already pluggable, upstream oauth2 should implement a filler interface underneath